### PR TITLE
Cabal gardening

### PR DESCRIPTION
**Pull request by michaelpj**
- Add `-Wunused-packages` to packages below `plutus-ledger-api` and
prune unused deps.
- Bump `source-repository-package` shas to match those used in
`cardano-ledger-specs`.
- Remove as many extra constraints from `cabal.project` as possible.
- Bumped hackage index-state.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by michaelpj** [comment]
Marking draft as I expect this to probably be broken somewhere and I'm going to let Hydra churn through it first.

**Comment by michaelpj** [comment]
I ended up wanting to bump `ouroboros-network`, I think it's all fine (tests pass) but @raduom might want to take a look.

File: plutus-pab/plutus-pab.cabal [hunk 1]
```diff
@@ -157,6 +157,7 @@ library
         network -any,
         network-mux -any,
         newtype-generics -any,
+        nothunks -any,
```
Comment (raduom) [inline plutus-pab/plutus-pab.cabal:160]: I was thinking that this is probably the successor of all the `NoUnexpectedThunks` classes.

**Review by raduom — APPROVED** [review]
Looks good to me. I am working on a PR in this area, so I can double check that everything is fine after you merge it.

File: plutus-pab/plutus-pab.cabal (hunk 1 above)
Comment (michaelpj) [reply plutus-pab/plutus-pab.cabal:160]: Yep, and we'll be using it elsewhere before long.

**Pull request merged**